apps/web/vibes/soul/primitives/alert/index.tsx

```diff
@@ -1,65 +1,81 @@
 import { clsx } from 'clsx';
 import { X } from 'lucide-react';
-import { ReactNode } from 'react';
 
 import { Button } from '@/vibes/soul/primitives/button';
 
-interface Props {
-  variant: 'success' | 'warning' | 'error' | 'info';
-  message: ReactNode;
-  description?: string;
-  dismissLabel?: string;
-  action?: {
-    label: string;
-    onClick: () => void;
-  };
-  onDismiss?: () => void;
-}
-
+/**
+ * This component supports various CSS variables for theming. Here's a comprehensive list, along
+ * with their default values:
+ *
+ * ```css
+ * :root {
+ *   --alert-success-background: color-mix(in oklab, hsl(var(--success)), white 75%);
+ *   --alert-warning-background: color-mix(in oklab, hsl(var(--warning)), white 75%);
+ *   --alert-error-background: color-mix(in oklab, hsl(var(--error)), white 75%);
+ *   --alert-info-background: hsl(var(--background));
+ *   --alert-font-family: var(--font-family-body);
+ *   --alert-border: hsl(var(--foreground) / 10%);
+ *   --alert-message-text: hsl(var(--foreground));
+ *   --alert-description-text: hsl(var(--foreground) / 50%);
+ * }
+ * ```
+ */
 export function Alert({
   variant,
   message,
   description,
   action,
   dismissLabel = 'Dismiss',
   onDismiss,
-}: Props) {
+}: {
+  variant: 'success' | 'warning' | 'error' | 'info';
+  message: string;
+  description?: string;
+  dismissLabel?: string;
+  action?: {
+    onClick: () => void;
+    text: string;
```

Comment:
I don't love using `text` here. Specially if the rationale is we're trying to avoid a conflict with `<label>` because it couples the prop interface to the internal implementation details. For example, imagine this `action` prop was actually used with a `<form>` as a React Action and so `action.text` was actually now used as a `<label>`? The idea that we're making decisions about the prop interface based on internal implementation details doesn't sit well with me